builtin/k8s/apply/platform.go

```diff
@@ -0,0 +1,302 @@
+package apply
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/docs"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Platform is the Platform implementation
+type Platform struct {
+	config Config
+}
+
+// Config implements Configurable
+func (p *Platform) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// DeployFunc implements component.Platform
+func (p *Platform) DeployFunc() interface{} {
+	return p.Deploy
+}
+
+// DestroyFunc implements component.Destroyer
+func (p *Platform) DestroyFunc() interface{} {
+	return p.Destroy
+}
+
+// GenerationFunc implements component.Generation
+func (p *Platform) GenerationFunc() interface{} {
+	return p.Generation
+}
+
+// Deploy deploys to Kubernetes
+func (p *Platform) Deploy(
+	ctx context.Context,
+	log hclog.Logger,
+	deployConfig *component.DeploymentConfig,
+	ui terminal.UI,
+) (*Deployment, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	s := sg.Add("")
+	defer func() { s.Abort() }()
+
+	deployment := &Deployment{
+		PruneLabel: p.config.PruneLabel,
+	}
+
+	s.Update("Executing kubectl apply...")
+	cmd, err := p.cmd(ctx, s, "apply", "-f", p.config.Path, "--prune", "-l", deployment.PruneLabel)
```

Comment:
It would be nice if  `-R` flag can be included, so that if there are any sub-directories, they can be applied too. Probably something similar during `Destroy`

Reply:
I'm not sure about this, but would it be a good idea to let users pass in any arguments kubectl supports. For instance, I can imagine someone wanting to supply `--server-side` flag during creation. During delete, user might want to supply the `--wait` flag.

Reply:
The `-R` flag is a good idea.

Not sure about the second one because it'd force us to be tied to `kubectl` instead of perhaps using just an API client one day (i.e. if server side apply becomes predominantly the way to go).

Reply:
Oh ok , and what about the `waypoint plan` issue? I couldn't find one for it. It might not work for all plugins but it does make sense for Kubernetes and Nomad. WDYT?

Reply:
There are no plans currently to do that, just because it would be hard to provide the same guarantees as Nomad and Terraform.